### PR TITLE
 Adding volume placement CRDs

### PR DESCRIPTION
**Pull request by harsh-px**
**What this PR does / why we need it**: This adds the volume placement CRDs. Portworx will register and watch these using https://github.com/portworx/sched-ops.

The CRD follows

- https://github.com/libopenstorage/openstorage/pull/665
- https://github.com/libopenstorage/openstorage/pull/685

The relevant changes are only in pkg/apis/portworx/v1beta. Rest of is compilation fixes for new vendors and rename from v1alpha1 to v1beta1 for the Cluster object.

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/apis/portworx/v1beta1/types.go [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+	Enforcement api.VolumePlacementRule_EnforcementType `json:"enforcement,omitempty"`
+	// Type is the type of the affinity rule
+	Type api.VolumePlacementRule_AffinityRuleType `json:"type,omitempty"`
+	// MatchExpressions is a list of label selector requirements. The requirements are ANDed.
+	MatchExpressions []*LabelSelectorRequirement `json:"matchExpressions,omitempty"`
+}
+
+// LabelSelectorRequirement A label selector requirement is a selector that contains values, a key, and an operator that
+// relates the key and values.
+type LabelSelectorRequirement struct {
```
Comment (piyush-nimbalkar) [inline pkg/apis/portworx/v1beta1/types.go]: should we use the LabelSelector object from k8s instead of defining on our side?

Comment (harsh-px) [reply pkg/apis/portworx/v1beta1/types.go]: Done

File: Gopkg.toml [hunk 2]
```diff
@@ -24,26 +24,28 @@
...
 [[constraint]]
-  name = "github.com/portworx/sched-ops"
+  name = "github.com/libopenstorage/openstorage"
   branch = "master"
 
 [[override]]
   name = "github.com/coreos/bbolt"
   version = "v1.3.1-coreos.6"
-
-[prune]
```
Comment (disrani-px) [inline Gopkg.toml]: The global prune should be there to prune unwanted files. You can add a per-project prune rule to not prune the code-generator files.

Comment (harsh-px) [reply Gopkg.toml]: Done

**Comment by harsh-px** [comment]
> Can you separate out the vendor changes from the first commit? Github isn't able to show the complete commit because of the large number of files.

PR updated with 4 separated commits now.

- new crd
- crd code generation script changes
- crd generated code
- vendoring changes

File: pkg/apis/portworx/register.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package portworx
+
+const (
+	// GroupName is the group name for the portworx CRD
+	GroupName = "portworx.com"
```
Comment (disrani-px) [inline pkg/apis/portworx/register.go]: Might want to use portworx.org or portworx.io instead

File: pkg/apis/portworx/v1beta1/register.go [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+import (
+	portworx "github.com/portworx/talisman/pkg/apis/portworx"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// SchemeGroupVersion is group version used to register these objects
+var SchemeGroupVersion = schema.GroupVersion{Group: portworx.GroupName, Version: "v1beta1"}
```
Comment (disrani-px) [inline pkg/apis/portworx/v1beta1/register.go]: portworx.Version for Version

Comment (harsh-px) [reply pkg/apis/portworx/v1beta1/register.go]: Done

File: pkg/apis/portworx/register.go (hunk 3 above)
Comment (harsh-px) [reply pkg/apis/portworx/register.go]: I kept portworx.com since it's resolvable. portworx.org or portworx.io will be dead links.

Comment (disrani-px) [reply pkg/apis/portworx/register.go]: That doesn't need to be resolvable, and we should probably get portworx.io anyways

Comment (harsh-px) [reply pkg/apis/portworx/register.go]: https://github.com/portworx/talisman/pull/69

**Pull request merged**